### DISCUSSION TITLE
Upgrading the reference architecture

### DISCUSSION THREAD

**Discussion by jeffreymlewis** [gruntwork-io/knowledge-base#588] [How do I]
Hello,

We rolled out the Reference Architecture in early 2021 and everything is good. I've noticed that the [example Reference Arch](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-production/infrastructure-live) has changed significantly since ours was deployed, and I'm wondering if there's any kind of upgrade guide? (Is the Reference Architecture versioned?)

I've seen guides like this, explaining how to [update to AWS provider v4](https://docs.gruntwork.io/guides/stay-up-to-date/terraform/how-to-update-to-aws-provider-v4), but I'm wondering if there's anything for upgrading the reference arch in infra-live (ie, new accounts.json format, the _envcommon directory, etc...)

Thanks,
Jeffrey

---

<ins datetime="2022-11-09T15:00:44Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109576">Tracked in ticket #109576</a></p>
</ins>

**Comment by ebeneliason** [comment]
Hi Jeffrey,

Some of the answers you're looking for can be found in [this article](https://github.com/gruntwork-io/knowledge-base/discussions/34). Feel free to follow up with any additional questions.
